include/boost/geometry/views/detail/boundary_view/implementation.hpp

```diff
@@ -0,0 +1,466 @@
+    }
+};
+
+
+template <typename Geometry, typename Tag = typename tag<Geometry>::type>
+struct views_container_initializer
+{};
+
+template <typename Polygon>
+struct views_container_initializer<Polygon, polygon_tag>
+{
+    template <typename BoundaryView>
+    static inline void apply(Polygon const& polygon, BoundaryView* views)
+    {
+        typedef polygon_rings_iterator<Polygon> rings_iterator_type;
+
+        std::uninitialized_copy(rings_iterator_type(polygon),
+                                rings_iterator_type(polygon, true),
+                                views);
+    }
+};
+
+template <typename MultiPolygon>
+class views_container_initializer<MultiPolygon, multi_polygon_tag>
+{
+    typedef typename boost::mpl::if_
+        <
+            boost::is_const<MultiPolygon>,
+            typename boost::range_value<MultiPolygon>::type const,
+            typename boost::range_value<MultiPolygon>::type
+        >::type polygon_type;
+
+    typedef polygon_rings_iterator<polygon_type> inner_iterator_type;
+
+    struct polygon_rings_begin
+    {
+        static inline inner_iterator_type apply(polygon_type& polygon)
+        {
+            return inner_iterator_type(polygon);
+        }
+    };
+
+    struct polygon_rings_end
+    {
+        static inline inner_iterator_type apply(polygon_type& polygon)
+        {
+            return inner_iterator_type(polygon, true);
+        }
+    };
+
+    typedef flatten_iterator
+        <
+            typename boost::range_iterator<MultiPolygon>::type,
+            inner_iterator_type,
+            typename std::iterator_traits<inner_iterator_type>::value_type,
+            polygon_rings_begin,
+            polygon_rings_end,
+            typename std::iterator_traits<inner_iterator_type>::reference
+        > rings_iterator_type;
+
+public:
+    template <typename BoundaryView>
+    static inline void apply(MultiPolygon const& multipolygon,
+                             BoundaryView* views)
+    {
+        rings_iterator_type first(boost::begin(multipolygon),
+                                  boost::end(multipolygon));
+        rings_iterator_type last(boost::end(multipolygon));
+
+        std::uninitialized_copy(first, last, views);
+    }
+};
+
+
+template <typename Areal>
+class areal_boundary
+{
+    typedef boundary_view<typename ring_type<Areal>::type> boundary_view_type;
+    typedef views_container_initializer<Areal> exception_safe_initializer;
+
+    template <typename T>
+    struct automatic_deallocator
+    {
+        automatic_deallocator(T* ptr) : m_ptr(ptr) {}
+
+        ~automatic_deallocator()
+        {
+            operator delete(m_ptr);
+        }
+
+        inline void release() { m_ptr = NULL; }
+
+        T* m_ptr;
+    };
+
+    inline void initialize_views(Areal const& areal)
+    {
+        // initialize number of rings
+        std::size_t n_rings = num_rings<Areal>::apply(areal);
+
+        if (n_rings == 0)
+        {
+            return;
+        }
```

Comment:
This is interesting. It implies that the empty range is [NULL, NULL+0). It should be fine I think.